manticore/core/worker.py

```diff
@@ -252,3 +263,138 @@ def start(self, target: typing.Optional[typing.Callable] = None):
+
+    def run(self, *args):
+        logger.debug(
+            "Capturing Logs via Thread %d. Pid %d Tid %d).",
+            self.id,
+            os.getpid(),
+            threading.get_ident(),
+        )
+
+        m = self.manticore
+        m._is_main = False
+
+        try:
+            with ReusableTCPServer((consts.HOST, consts.PORT), DumpTCPHandler) as server:
+                server.dump = self.dump_logs  # type: ignore
+                server.serve_forever()
+        except OSError as e:
+            # TODO - this should be logger.warning, but we need to rewrite several unit tests that depend on
+            # specific stdout output in order to do that.
+            logger.info("Could not start log capture server: %s", str(e))
+
+
+def render_state_descriptors(desc: typing.Dict[int, StateDescriptor]):
+    """
+    Converts the built-in list of state descriptors into a StateList from Protobuf
+
+    :param desc: Output from ManticoreBase.introspect
+    :return: Protobuf StateList to send over the wire
+    """
+    out = StateList()
+    for st in desc.values():
+        if st.status != StateStatus.destroyed:
+            now = datetime.now()
+            out.states.append(
+                State(
+                    id=st.state_id,
+                    type={
+                        StateLists.ready: State.READY,  # type: ignore
+                        StateLists.busy: State.BUSY,  # type: ignore
+                        StateLists.terminated: State.TERMINATED,  # type: ignore
+                        StateLists.killed: State.KILLED,  # type: ignore
+                    }[
+                        getattr(st, "state_list", StateLists.killed)
+                    ],  # If the state list is missing, assume it's killed
+                    reason=st.termination_msg,
+                    num_executing=st.own_execs,
+                    wait_time=int(
+                        (now - st.field_updated_at.get("state_list", now)).total_seconds() * 1000
+                    ),
+                )
+            )
+    return out
+
+
+def state_monitor(self: DaemonThread):
+    """
+    Daemon thread callback that runs a server that listens for incoming TCP connections and
+    dumps the list of state descriptors.
+
+    :param self: DeamonThread created to run the server
+    """
+    logger.debug(
+        "Monitoring States via Thread %d. Pid %d Tid %d).",
+        self.id,
+        os.getpid(),
+        threading.get_ident(),
+    )
+
+    m = self.manticore
+
+    def dump_states():
+        sts = m.introspect()
+        sts = render_state_descriptors(sts)
+        return sts.SerializeToString()
+
+    try:
+        with ReusableTCPServer((consts.HOST, consts.PORT + 1), DumpTCPHandler) as server:
+            server.dump = dump_states  # type: ignore
+            server.serve_forever()
+    except OSError as e:
+        # TODO - this should be logger.warning, but we need to rewrite several unit tests that depend on
```

Comment:
@feliam Do you think it's worth fixing this in this PR, or should we do a more general rewrite of the tests separately? 

A lot of our tests check for exact stdout output, which can be brittle if you want to add new logging messages. Maybe it'd be better to check that the lines we're looking for are contained in the output the correct number of times?